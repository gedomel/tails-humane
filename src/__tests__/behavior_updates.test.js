import React from 'react';
import renderer from 'react-test-renderer';
import UpdateForm from '../Components/update_form';

/**
 * Test Method to ensure dropdown 'No Kids' selection updates as expected
 */
it('Render An Updated No Kids Status Update', () => {
  const tree = renderer
    .create(
      <UpdateForm
        name="Mango"
        animalID="395468"
        availablility="Available"
        housetrained="Yes"
        noDogs="Unknown"
        noCats="Unknown"
        noKids="OK"
        species="Dog"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/**
 * Test Method to ensure dropdown 'No Dogs' selection updates as expected
 */
it('Render A No Dogs Status Update Correctly', () => {
  const tree = renderer
    .create(
      <UpdateForm
        name="Mr. Man"
        animalID="7888890"
        availablility="Available"
        housetrained="Unknown"
        noDogs="No dogs"
        noCats="Unknown"
        noKids="Unknown"
        species="Cat"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/**
 * Test Method to ensure dropdown 'No Cats' selection updates as expected
 */
it('Render A No Cats Status Update Correctly', () => {
  const tree = renderer
    .create(
      <UpdateForm
        name="Mango"
        animalID="395468"
        availablility="Available"
        housetrained="Yes"
        noDogs="Unknown"
        noCats="No cats"
        noKids="Unknown"
        species="Dog"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
