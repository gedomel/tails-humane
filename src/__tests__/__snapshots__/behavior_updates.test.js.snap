// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render A No Cats Status Update Correctly 1`] = `
<div
  className="updateform"
>
  <form
    onSubmit={[Function]}
  >
    <p>
      Name:
      Mango
    </p>
    <p>
      TAILS ID:
      395468
    </p>
    <p>
      Species:
      Dog
    </p>
    <p>
      Select Availability:
    </p>
    <div>
      <select
        onChange={[Function]}
        value="Available"
      >
        <option
          value="Available"
        >
          Available
        </option>
        <option
          value="Unavailable"
        >
          Unavailable
        </option>
      </select>
      <p>
        Housetrained?:
      </p>
      <select
        onChange={[Function]}
        value="Yes"
      >
        <option
          value="Partially"
        >
          Partially
        </option>
        <option
          value="Yes"
        >
          Yes
        </option>
        <option
          value="No"
        >
          No
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Dogs?:
      </p>
      <select
        onChange={[Function]}
        value="Unknown"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No dogs"
        >
          No dogs
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Cats?:
      </p>
      <select
        onChange={[Function]}
        value="No cats"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No cats"
        >
          No cats
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Kids?:
      </p>
      <select
        onChange={[Function]}
        value="Unknown"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No small kids"
        >
          No small kids
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
    </div>
    <button
      className="SubmitButton"
      type="submit"
    >
      Submit Changes
    </button>
    <p
      className="success"
    >
      
    </p>
  </form>
</div>
`;

exports[`Render A No Dogs Status Update Correctly 1`] = `
<div
  className="updateform"
>
  <form
    onSubmit={[Function]}
  >
    <p>
      Name:
      Mr. Man
    </p>
    <p>
      TAILS ID:
      7888890
    </p>
    <p>
      Species:
      Cat
    </p>
    <p>
      Select Availability:
    </p>
    <div>
      <select
        onChange={[Function]}
        value="Available"
      >
        <option
          value="Available"
        >
          Available
        </option>
        <option
          value="Unavailable"
        >
          Unavailable
        </option>
      </select>
      <p>
        Housetrained?:
      </p>
      <select
        onChange={[Function]}
        value="Unknown"
      >
        <option
          value="Partially"
        >
          Partially
        </option>
        <option
          value="Yes"
        >
          Yes
        </option>
        <option
          value="No"
        >
          No
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Dogs?:
      </p>
      <select
        onChange={[Function]}
        value="No dogs"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No dogs"
        >
          No dogs
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Cats?:
      </p>
      <select
        onChange={[Function]}
        value="Unknown"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No cats"
        >
          No cats
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Kids?:
      </p>
      <select
        onChange={[Function]}
        value="Unknown"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No small kids"
        >
          No small kids
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
    </div>
    <button
      className="SubmitButton"
      type="submit"
    >
      Submit Changes
    </button>
    <p
      className="success"
    >
      
    </p>
  </form>
</div>
`;

exports[`Render An Updated No Kids Status Update 1`] = `
<div
  className="updateform"
>
  <form
    onSubmit={[Function]}
  >
    <p>
      Name:
      Mango
    </p>
    <p>
      TAILS ID:
      395468
    </p>
    <p>
      Species:
      Dog
    </p>
    <p>
      Select Availability:
    </p>
    <div>
      <select
        onChange={[Function]}
        value="Available"
      >
        <option
          value="Available"
        >
          Available
        </option>
        <option
          value="Unavailable"
        >
          Unavailable
        </option>
      </select>
      <p>
        Housetrained?:
      </p>
      <select
        onChange={[Function]}
        value="Yes"
      >
        <option
          value="Partially"
        >
          Partially
        </option>
        <option
          value="Yes"
        >
          Yes
        </option>
        <option
          value="No"
        >
          No
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Dogs?:
      </p>
      <select
        onChange={[Function]}
        value="Unknown"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No dogs"
        >
          No dogs
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Cats?:
      </p>
      <select
        onChange={[Function]}
        value="Unknown"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No cats"
        >
          No cats
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Kids?:
      </p>
      <select
        onChange={[Function]}
        value="OK"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No small kids"
        >
          No small kids
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
    </div>
    <button
      className="SubmitButton"
      type="submit"
    >
      Submit Changes
    </button>
    <p
      className="success"
    >
      
    </p>
  </form>
</div>
`;
